indexer/src/main.rs

```diff
@@ -0,0 +1,186 @@
+    );
+
+    // Accounts
+    let accounts_future = async {
+        let futures = streamer_message.shards.iter().map(|shard| {
+            adapters::accounts::handle_accounts(
+                pool,
+                &shard.receipt_execution_outcomes,
+                streamer_message.block.header.height,
+            )
+        });
+
+        try_join_all(futures).await.map(|_| ())
+    };
+
+    // Event-based entities (FT, NFT)
+    let assets_events_future = adapters::assets::events::store_events(pool, &streamer_message);
+
+    if strict_mode {
+        // AccessKeys
+        let access_keys_future = async {
+            let futures = streamer_message.shards.iter().map(|shard| {
+                adapters::access_keys::handle_access_keys(
+                    pool,
+                    &shard.state_changes,
+                    streamer_message.block.header.height,
+                )
+            });
+
+            try_join_all(futures).await.map(|_| ())
+        };
+
+        // StateChange related to Account
+        let account_changes_future = adapters::account_changes::store_account_changes(
+            pool,
+            &streamer_message.shards,
+            &streamer_message.block.header.hash,
+            streamer_message.block.header.timestamp,
+        );
+
+        try_join!(
+            execution_outcomes_future,
+            accounts_future,
+            access_keys_future,
+            assets_events_future,
+            account_changes_future,
+        )?;
+    } else {
+        try_join!(
+            execution_outcomes_future,
+            accounts_future,
+            assets_events_future
+        )?;
+    }
+
+    Ok(())
+}
+
+#[actix::main]
+async fn main() -> anyhow::Result<()> {
+    // We use it to automatically search the for root certificates to perform HTTPS calls
+    // (sending telemetry and downloading genesis)
+    openssl_probe::init_ssl_cert_env_vars();
+
+    dotenv::dotenv().ok();
+
+    let opts: Opts = Opts::parse();
+
+    configs::init_tracing(opts.debug)?;
+
+    // We establish connection as early as possible as an additional sanity check.
+    // Indexer should fail if .env file with credentials is missing/wrong
+    let pool = models::establish_connection(&opts.database_url);
+
+    let strict_mode = !opts.non_strict_mode;
+
+    // We want to prevent unnecessary SELECT queries to the database to find
+    // the Transaction hash for the Receipt.
+    // Later we need to find the Receipt which is a parent to underlying Receipts.
+    // Receipt ID will of the child will be stored as key and parent Transaction hash/Receipt ID
+    // will be stored as a value
+    let receipts_cache: receipts_cache::ReceiptsCache =
+        std::sync::Arc::new(Mutex::new(SizedCache::with_size(100_000)));
+
+    let config: near_lake_framework::LakeConfig = opts.to_lake_config().await;
+    let (_, stream) = near_lake_framework::streamer(config);
+
+    tokio::spawn(async move {
```

Comment:
Could somebody remind me how it comes we use `[actix::main]` but `tokio::spawn`?

cc @frol @telezhnaya

Reply:
@khorolets It was recently introduced when metrics were added as either streaming or metrics need to be spawned (alternatively, they can be `select!`ed, but it does not make a lot of sense)

Reply:
Maybe it is worth switching to `[tokio::main]`?

Reply:
Am I right that we use actix only because of diesel?
Anyway, I'm with @khorolets , let's use `[tokio::main]`

Reply:
@telezhnaya yeah, I recall tokio-diesel didn't work (or stopped working), we can try to change it later, separately.